build-gluster-org/scripts/user-update.py

```diff
@@ -0,0 +1,99 @@
+import requests
+import re
+import os
+from git import Repo
+import subprocess
+
+#Collects the gluster-maintainers names
+def create_user_data(repo_path):
+    os.chdir(repo_path)
+    username = "rakshithakamath94"
+    #Token will pe passed as secret text from jenkins
+    token = os.environ['TOKEN_RK']
+    user_data = requests.get('https://api.github.com/orgs/gluster/teams/gluster-maintainers/members', headers={'Authorization': 'token {}'.format(token)}).json()
+    return user_data
+
+#Function to find out the names which are missing in the jenkins conf file
+def search_pat(repo_path, filename, string_to_search,user_data):
+    os.chdir(repo_path)
+    INDENTATION_VALUE = ""
+    NOT_FOUND = []
+    for i, line in enumerate(open(filename)):
+        if string_to_search in line:
+            INDENTATION_VALUE=re.match(r"\s*", line).group()
+    for username in user_data:
+        with open(filename) as f:
+            if re.search(username['login'], f.read()):
+                continue
+            else:
+                NOT_FOUND.append("%s  "%INDENTATION_VALUE+"- "+username['login'])
+    f.close()
+    return NOT_FOUND
+
+#Function which appends the names which are not present in the jenkins conf file
+def edit_file(repo_path,filename,pattern,NOT_FOUND):
+    os.chdir(repo_path)
+    os.getcwd()
+    os.system('ls')
+    with open(filename, "r") as input_file:
+        buffer = input_file.readlines()
+
+    with open(filename, "w") as output_file:
+        search_pattern = re.compile(r'\s*%s'%pattern) 
```

Comment:
wouldn't it be more robust to parse the yaml file using yaml module, and then write it down again with  dump ?

Reply:
I thought a simple file operation would do our job but your suggestion is valid and I would take a look at the same, I will try to update the script with the appropriate changes